### PR TITLE
[Mobile]: More functionalities

### PR DESCRIPTION
**Pull request by joel-jeremy**
Added for mobile:
- Updated the balance tooltip to be a modal
- Scheduled transaction action modals
  - Post transaction
  - Skip scheduled date
  - View schedule (not supported for now)
- Budget summary action modals
  - Move to a category
  - Hold for next month
  - Reset hold buffer
- Account action modals
  - Rename account
  - Close account
  - Reopen account

### CODE REVIEW

**Comment by MatissJanis** [comment]
This is a massive amount of LOC. I started reviewing, but quickly got lost. Any chance you could break this out in smaller PRs?

**Comment by joel-jeremy** [comment]
> This is a massive amount of LOC. I started reviewing, but quickly got lost. Any chance you could break this out in smaller PRs?

Sorry for the big PR. I had some extra time so I tried doing as much as I can before I get a bit busy again. Any tips how to break this down into smaller chunks?

For some of the changes, I have:
- Extracted some `Menu` components used in desktop e.g. BalanceMenu, ToBudgetMenu to their own files so that they can be reused in mobile.
- Made some changes to the Autocomplete component to fix the typecheck errors that I was encountering.
- Extracted category/payee/account autocomplete modals from EditField to their own files/components

File: packages/desktop-client/src/components/Modals.tsx [hunk 1]
```diff
@@ -281,17 +335,22 @@ export function Modals() {
 
         case 'rollover-budget-summary':
           return (
-            <RolloverBudgetSummary
+            <NamespaceContext.Provider
```
Comment (joel-jeremy) [inline packages/desktop-client/src/components/Modals.tsx:338]: We need to think about how to remove the need to wrap modal components with this in order for `useSheetValue` to work. By default, there is no namespace context for any of our modals.

**Comment by MatissJanis** [comment]
> > This is a massive amount of LOC. I started reviewing, but quickly got lost. Any chance you could break this out in smaller PRs?
> 
> Sorry for the big PR. I had some extra time so I tried doing as much as I can before I get a bit busy again. Any tips how to break this down into smaller chunks?
> 
> For some of the changes, I have:
> 
> * Extracted some `Menu` components used in desktop e.g. BalanceMenu, ToBudgetMenu to their own files so that they can be reused in mobile.
> * Made some changes to the Autocomplete component to fix the typecheck errors that I was encountering.
> * Extracted category/payee/account autocomplete modals from EditField to their own files/components

This list is a great starting point for the breaking-up into smaller chunks. Especially the bigger changes such as file/component moves (with no logic changes) could be extracted to separate PRs for a big drop in LOC here.

**Comment by joel-jeremy** [comment]
@MatissJanis I've split the autocomplete changes here: https://github.com/actualbudget/actual/pull/2500

**Comment by joel-jeremy** [comment]
Split menu components here: https://github.com/actualbudget/actual/pull/2511

**Comment by joel-jeremy** [comment]
@MatissJanis This should now be ready for review :) Most related refactors have been merged to master.

File: packages/desktop-client/src/components/modals/CoverModal.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+  useEffect(() => {
+    if (initialMount) {
+      onCategoryClick();
+    }
+  }, [initialMount, onCategoryClick]);
+
+  const category = categories.find(c => c.id === categoryId);
+
+  if (category == null) {
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/modals/CoverModal.tsx:64]: **🥜 nitpick:** 

```suggestion
  if (category === null) {
```

**Review by MatissJanis — COMMENTED** [review]
Overall LGTM. Left a few comments. Let me know what you think.

**Review by MatissJanis — APPROVED** [review]

**Pull request merged**